includes/class-newspack-newsletters-contacts.php

```diff
@@ -0,0 +1,257 @@
+			Newspack_Newsletters_Subscription::add_subscription_intent( $contact, $lists );
+			return true;
+		}
+
+		if ( false !== $lists ) {
+			Newspack_Newsletters_Logger::log( 'Adding contact to list(s): ' . implode( ', ', $lists ) . '. Provider is ' . $provider->service . '.' );
+		} else {
+			Newspack_Newsletters_Logger::log( 'Adding contact without lists. Provider is ' . $provider->service . '.' );
+		}
+
+		$existing_contact                 = Newspack_Newsletters_Subscription::get_contact_data( $contact['email'], true );
+		$contact['existing_contact_data'] = \is_wp_error( $existing_contact ) ? false : $existing_contact;
+		$is_updating                      = \is_wp_error( $existing_contact ) ? false : true;
+
+		/**
+		 * Filters the contact before passing on to the API.
+		 *
+		 * @param array          $contact           {
+		 *          Contact information.
+		 *
+		 *    @type string   $email                 Contact email address.
+		 *    @type string   $name                  Contact name. Optional.
+		 *    @type string   $existing_contact_data Existing contact data, if updating a contact. The hook will be also called when
+		 *    @type string[] $metadata              Contact additional metadata. Optional.
+		 * }
+		 * @param string[]|false $selected_list_ids Array of list IDs the contact will be subscribed to, or false.
+		 * @param string         $provider          The provider name.
+		 */
+		$contact = apply_filters( 'newspack_newsletters_contact_data', $contact, $lists, $provider->service );
+
+		if ( isset( $contact['metadata'] ) ) {
+			Newspack_Newsletters_Logger::log( 'Adding contact with metadata key(s): ' . implode( ', ', array_keys( $contact['metadata'] ) ) . '.' );
+		}
+
+		if ( ! isset( $contact['metadata'] ) ) {
+			$contact['metadata'] = [];
+		}
+		$contact['metadata']['origin_newspack'] = '1';
+
+		/**
+		 * Filters the contact selected lists before passing on to the API.
+		 *
+		 * @param string[]|false $lists    Array of list IDs the contact will be subscribed to, or false.
+		 * @param array          $contact  {
+		 *          Contact information.
+		 *
+		 *    @type string   $email    Contact email address.
+		 *    @type string   $name     Contact name. Optional.
+		 *    @type string[] $metadata Contact additional metadata. Optional.
+		 * }
+		 * @param string         $provider          The provider name.
+		 */
+		$lists = apply_filters( 'newspack_newsletters_contact_lists', $lists, $contact, $provider->service );
+
+		return self::add_to_esp( $contact, $lists, $is_updating );
+	}
+
+	/**
+	 * Permanently delete a user subscription.
+	 *
+	 * @param int $user_id User ID.
+	 *
+	 * @return bool|WP_Error Whether the contact was deleted or error.
+	 */
+	public static function delete( $user_id ) {
+		$user = get_user_by( 'id', $user_id );
+		if ( ! $user ) {
+			return new WP_Error( 'newspack_newsletters_invalid_user', __( 'Invalid user.' ) );
+		}
+		/** Only delete if email ownership is verified. */
```

Comment:
Non-blocking question: Do we want this to be the case? If deleting a user account via admin, then unverified reader accounts will be deleted in the site but not in the ESP, creating a data discrepancy. Non-blocking because this is currently true in `trunk` as well.

Reply:
Yes, agree with you, it looks odd.

But we are not changing behavior in the PR. If we want to change that I'd say let's do it another time